llama_index/output_parsers/selection.py

```diff
@@ -48,30 +48,24 @@ class Answer(DataClassJsonMixin):
 
 
 class SelectionOutputParser(BaseOutputParser):
-    REQUIRED_KEYS = {"choice", "reason"}
+    REQUIRED_KEYS = frozenset(Answer.__annotations__)
 
     def _filter_dict(self, json_dict: dict) -> dict:
+        """Filter recursively until a dictionary matches all REQUIRED_KEYS."""
         output_dict = json_dict
         for key, val in json_dict.items():
             if key in self.REQUIRED_KEYS:
                 continue
             elif isinstance(val, dict):
-                found = True
```

Comment:
This is like half baked recursion, we should just do full-blown recursion

Reply:
lol thanks for refactoring my spaghetti code